src/main/java/dinkplugin/notifiers/KillCountNotifier.java

```diff
@@ -76,16 +104,54 @@ private void handleKill(String boss, int killCount, String rawMessage) {
         createMessage(screenshot, body.build());
     }
 
-    private boolean checkKillInterval(int killCount) {
+    private boolean checkKillInterval(int killCount, @Nullable Boolean pb) {
+        if (pb == Boolean.TRUE && config.killCountNotifyBestTime())
+            return true;
+
         if (killCount == 1 && config.killCountNotifyInitial())
             return true;
 
         int interval = config.killCountInterval();
         return interval <= 1 || killCount % interval == 0;
     }
 
+    private void updateData(BossNotificationData updated) {
+        if (data == null) {
+            this.data = updated;
+        } else {
+            // Boss data and timing are sent in separate messages
+            // where the order of the messages differs depending on the boss.
+            // Here, we update data without setting any not-null values back to null.
+            this.data = new BossNotificationData(
+                defaultIfNull(updated.getBoss(), data.getBoss()),
+                defaultIfNull(updated.getCount(), data.getCount()),
+                defaultIfNull(updated.getGameMessage(), data.getGameMessage()),
+                defaultIfNull(updated.getTime(), data.getTime()),
+                defaultIfNull(updated.isPersonalBest(), data.isPersonalBest())
+            );
+        }
+    }
+
+    private static Optional<BossNotificationData> parse(String message) {
+        // try to parse kill count or fight duration
+        return parseBoss(message)
+            .map(pair -> new BossNotificationData(pair.getLeft(), pair.getRight(), message, null, null))
+            .map(Optional::of) // hack since Optional#or was added in jdk 9, not 8
+            .orElseGet(() -> parseTime(message).map(time -> new BossNotificationData(null, null, null, time.getLeft(), time.getRight())));
+    }
```

Comment:
I feel like this would be clearer to write as two if-cases

something like

```
if (auto pair = parseBoss(message)) {
  ... return parse boss data
}

if (auto pair = parseTime)) {
  ... return parse time data
}

return none
```